eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/VariantExporter.java

```diff
@@ -87,227 +62,26 @@
-        return getVariantStart(maxQuery);
-    }
-
-    private List<Region> getRegionsFromQuery(String regionFilter) {
-        if (regionsInFilter == null) {
-            regionsInFilter = Region.parseRegions(regionFilter);
-        }
-        return regionsInFilter;
-    }
-
-
-
-    private List<Region> getRegionsForChromosome(String chromosome, int chromosomeMinStart, int chromosomeMaxStart) {
-        List<Region> regions = regionDivider.divideRegionInChunks(chromosome, chromosomeMinStart, chromosomeMaxStart);
-        logger.debug("Number of regions in chromosome{}: {}", chromosome, regions.size());
-        if (!regions.isEmpty()) {
-            logger.debug("First region: {}", regions.get(0));
-            logger.debug("Last region: {}", regions.get(regions.size() - 1));
-        }
-
-        return regions;
-    }
 
-    private List<Region> getRegionsListForQueryRegions(List<Region> regionsFromQuery) {
-        List<Region> regions = new ArrayList<>();
-        for (Region region : regionsFromQuery) {
-            regions.addAll(regionDivider.divideRegionInChunks(region));
-        }
-        return regions;
-    }
-
-    private QueryOptions addChromosomeSortAndLimitToQuery(String chromosome, boolean ascending) {
-        QueryOptions chromosomeSortedByStartQuery = new QueryOptions(options);
-        chromosomeSortedByStartQuery.put(VariantDBAdaptor.CHROMOSOME, chromosome);
-
-        BasicDBObject sortDBObject = new BasicDBObject();
-        int orderOperator = ascending ? 1 : -1;
-        sortDBObject.put("chr", orderOperator);
-        sortDBObject.put("start", orderOperator);
-        chromosomeSortedByStartQuery.put("sort", sortDBObject);
-
-        chromosomeSortedByStartQuery.put("limit", 1);
-
-        return chromosomeSortedByStartQuery;
-    }
+        // region sequence contains the last exported region: we set it to null to get the new region sequence from cellbase if needed
+        regionSequence = null;
 
-    private int getVariantStart(QueryOptions query) {
-        int start = -1;
-        VariantDBIterator variantDBIterator = variantDBAdaptor.iterator(query);
-        if (variantDBIterator.hasNext()) {
-            Variant variant = variantDBIterator.next();
-            start = variant.getStart();
-        }
-
-        return start;
-    }
-
-
-    private Map<String, List<VariantContext>> dumpVariantsInRegion(Region region, Map<String, VariantSource> sources) throws IOException {
-        Map<String, List<VariantContext>> dumpedVariants = new HashMap<>();
-        QueryOptions regionQuery = addRegionToQuery(region);
-        regionQuery = addProjectionToQuery(regionQuery);
-        VariantDBIterator variantDBIterator = variantDBAdaptor.iterator(regionQuery);
-        while (variantDBIterator.hasNext()) {
-            Variant variant = variantDBIterator.next();
+        while (iterator.hasNext()) {
+            Variant variant = iterator.next();
```

Comment:
To use a "for each" loop we need a class that implements "Iterable", and VariantDBIterator doesn't. VariantMongoDBAdaptor is "iterable", but if we use the for each loop over it, it will use the iterator() method with no parameters, that ignores the filters and options from QueryOptions. 

We can change this for a traditional for method: 
for (Variant v; v=variantIterator.next(); variantIterator.hasNext())

Do I change this?